alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/action/DeleteDestinationAction.kt

```diff
@@ -0,0 +1,27 @@
+/*
+ *   Copyright 2020 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License").
+ *   You may not use this file except in compliance with the License.
+ *   A copy of the License is located at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   or in the "license" file accompanying this file. This file is distributed
+ *   on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ *   express or implied. See the License for the specific language governing
+ *   permissions and limitations under the License.
+ */
+
+package com.amazon.opendistroforelasticsearch.alerting.action
+
+import org.elasticsearch.action.ActionType
+import org.elasticsearch.action.delete.DeleteResponse
+import org.elasticsearch.common.io.stream.Writeable
+
+class DeleteDestinationAction private constructor() : ActionType<DeleteResponse>(NAME, Writeable.Reader({ DeleteResponse(null) })) {
```

Comment:
If we wish to make `DeleteDestinationAction` itself a singleton, is it possible to define this as an `object`, instead of using a `companion object` to store `INSTANCE`?

Reply:
Thanks. Discussed with Drew also, ISM currently using companion object. Keeping as it is for consistency.